### DISCUSSION TITLE
Sending emails with telnet without authentication

### DISCUSSION THREAD

**Discussion by JadKHaddad** [postalserver/postal#2662] [Help with using Postal] [answered]
I was able to send emails using telnet without any kind of authentication and this feels wrong. Is there a way to enforce authentication?

**Comment by willpower232** [comment]
Did you definitely see `ESMTP Postal` at the top of your telnet output?

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/lib/postal/smtp_server/client.rb#L123

**Comment by JadKHaddad** [comment]
I came up with these three scenarios

```powershell
telnet postal.domain.com 25

220 postal.domain.com ESMTP Postal/DWSDRF
HELO
250 postal.domain.com
MAIL FROM: support@domain.com
250 OK
RCPT TO: someone@gmail.com
530 Authentication required
```

```powershell
telnet postal.domain.com 25

220 postal.domain.com ESMTP Postal/F4UQMF
EHLO
250-My capabilities are
250-STARTTLS
250 AUTH CRAM-MD5 PLAIN LOGIN
auth login
334 VXNlcm5hbWU6
c29tZXRoaW5n
334 UGFzc3dvcmQ6
c29tZXRoaW5n
235 Granted for domain-2/main
HELO
250 postal.domain.com
MAIL FROM: support@domain.com
250 OK
RCPT TO: someone@gmail.com
250 OK
DATA
354 Go ahead
From: support@domain.com
To: someone@gmail.com
Subject: Test

test auth

.
250 OK
QUIT
221 Closing Connection
```

```powershell
telnet postal.domain.com 25

220 postal.domain.com ESMTP Postal/K80UPW
HELO
250 postal.domain.com
MAIL FROM: someone@gmail.com
250 OK
RCPT TO: support@domain.com
250 OK
DATA
354 Go ahead
From: someone@gmail.com
To: support@domain.com
Subject: Test

test

.
250 OK
QUIT
221 Closing Connection
```

The first two scenarios make sense, but the third one doesn't. The third email is being sent from ```someone@gmail.com``` and I can see it under ```Incoming Messages``` in the web-UI. Do I have to verify manually, that the email really comes from ```someone@gmail.com```? with like a signiture or someting? Sorry if I'm being dumb, but I'm new to this.

**Reply by willpower232** [reply]
from your example, if Postal knows "domain.com" then it will receive the message briefly to determine if it can handle it via a route, did the message appear in the UI for the Server that has that domain?

**Comment by JadKHaddad** [comment]
Yes sir. And it looks like this.

![Screenshot-postal](https://github.com/postalserver/postal/assets/55059802/75430355-0525-42d5-8f07-93a22352e1ef)

**Reply by willpower232** [reply]
> message has been accepted but not sent to any endpoints

sounds like its working to me

**Comment by JadKHaddad** [comment]
I mean ```someone@gmail.com``` obviously  did not send the email. So this is normal?

**Reply by willpower232** [reply, marked as answer]
If you had an SMTP route set up for example then whatever eventually received the message would check the headers against gmail.com and see that the message was not legitimate.

I think you can also enable the spamassassin/clamav/rspamd integrations to check the incoming messages but Postal is always just a messenger I'm afraid.

**Reply by JadKHaddad** [reply]
thank you!
